crates/matrix-sdk/src/oidc.rs

```diff
@@ -781,6 +851,23 @@ pub(crate) struct OidcData {
     pub(crate) latest_id_token: RwLock<Option<IdToken<'static>>>,
 }
 
+/// Information about the URL that should be presented to authenticate with the
+/// server.
+#[derive(Debug)]
+pub struct OidcAuthenticationUrl {
+    /// The URL that should be presented.
+    pub url: Url,
+    /// Data necessary to validate a response from the endpoint.
+    pub validation_data: AuthorizationValidationData,
```

Comment:
It appears to me that this is the only data needed to verify that the configuration is still correct for the login to complete, but if not I'm happy to add other state here too which we could compare before completing the login.

Reply:
This was never meant to be a public property, it should have been private so that the clients couldn't access it. That said I had a go at the hashmap today but it was always empty.

> clients need to keep the Oidc struct around at least until the authorization returns

I now understand this comment 🙈

Will take another look at it tomorrow.